src/mds/MDCache.cc

```diff
@@ -1414,6 +1425,38 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Default cache memory limit
+
+
+/*
+ * Get the default cache memory limit using the mds_memory_target
+ */
+uint64_t MDCache::get_default_mds_cache_memory_limit()
+{
+  uint64_t temp_cache_limit = (uint64_t) (mds->memory_target/1.25);
+  mds->cct->_conf.set_val_default("mds_cache_memory_limit", stringify(temp_cache_limit));
+  return temp_cache_limit;
+}
+
+// ===================================
+// Cache Autotuning
+
+
+/*
+ * If mds_cache_autotune is set to true, then adjust cache_memory_limit using the rss usage periodically as per the mds_cache_resize_interval
+ */
+void MDCache::resize()
+{
+    MemoryModel::snap last;
+    mm.sample(&last);
+    if (((uint64_t)(last.get_rss())) > mds->memory_target) 
+      cache_memory_limit *= 0.99;
```

Comment:
Also need to check if `cache_memory_limit > mds->memory_target`. If that happens, it's because someone has changed `mds->memory_target` externally. In that case, we should do `cache_memory_limit = mds->memory_target / 1.25`.

Also add a comment for 1.25 value.